src/main/java/co/cask/gcp/speech/SpeechToTextTransform.java

```diff
@@ -56,79 +56,72 @@
   public static final String NAME = "SpeechToText";
   public static final String DESCRIPTION = "Converts audio files to text by applying powerful neural network models.";
   private SpeechTransformConfig config;
-  private Schema schema;
+  private Schema outputSchema = null;
   private RecognitionConfig recognitionConfig;
   private SpeechClient speech;
 
   @Override
   public void configurePipeline(PipelineConfigurer configurer) throws IllegalArgumentException {
     super.configurePipeline(configurer);
-    Schema inputSchema = configurer.getStageConfigurer().getInputSchema();
-    // If a inputSchema and audioField is available then verify that the schema does contain the given audioField
-    // and that the type is byte array. This will allow to fail fast i.e. during deployment time.
-    if (inputSchema != null && !config.containsMacro("audiofield")) {
-      if (inputSchema.getField(config.audioField) == null) {
-        throw new IllegalArgumentException(String.format("Field %s must exists.", config.audioField));
-      }
-      Schema.Field field = inputSchema.getField(config.audioField);
-      Schema fieldSchema = field.getSchema();
-      Schema.Type fieldType = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
-      if (fieldType != Schema.Type.BYTES) {
-        throw new IllegalArgumentException(String.format("Field '%s' must be of type 'bytes' but is of type " +
-                                                           "'%s'.", field.getName(), fieldType));
-      }
-    }
+    config.validate(configurer.getStageConfigurer().getInputSchema());
   }
 
   @Override
   public void initialize(TransformContext context) throws Exception {
     super.initialize(context);
-    schema = config.getSchema();
+    config.validate(context.getInputSchema());
```

Comment:
don't need to validate in initialize(), but should validate in prepareRun(), which is currently not overridden. prepareRun() is before the run starts and before an initialize() call is made